From 05a48f0ebc39c13a1fc5c67ed48a94b6de7c3778 Mon Sep 17 00:00:00 2001
From: Lorenzo Colitti <lorenzo@google.com>
Date: Thu, 27 Nov 2014 15:12:10 +0900
Subject: [PATCH 501/634] Make suspend abort reason logging depend on
 CONFIG_PM_SLEEP

This unbreaks the build on architectures such as um that do not
support CONFIG_PM_SLEEP.

Change-Id: Ia846ed0a7fca1d762ececad20748d23610e8544f
Signed-off-by: Lorenzo Colitti <lorenzo@google.com>
---
 kernel/power/process.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kernel/power/process.c b/kernel/power/process.c
index 4c83e52..78915a3 100644
--- a/kernel/power/process.c
+++ b/kernel/power/process.c
@@ -34,7 +34,9 @@ static int try_to_freeze_tasks(bool user_only)
 	unsigned int elapsed_msecs;
 	bool wakeup = false;
 	int sleep_usecs = USEC_PER_MSEC;
+#ifdef CONFIG_PM_SLEEP
 	char suspend_abort[MAX_SUSPEND_ABORT_LEN];
+#endif
 
 	do_gettimeofday(&start);
 
@@ -64,9 +66,11 @@ static int try_to_freeze_tasks(bool user_only)
 			break;
 
 		if (pm_wakeup_pending()) {
+#ifdef CONFIG_PM_SLEEP
 			pm_get_active_wakeup_sources(suspend_abort,
 				MAX_SUSPEND_ABORT_LEN);
 			log_suspend_abort_reason(suspend_abort);
+#endif
 			wakeup = true;
 			break;
 		}
-- 
1.9.1

